compiler/rustc_serialize/src/opaque.rs

```diff
@@ -193,7 +193,7 @@ impl FileEncoder {
         // shaves an instruction off those code paths (on x86 at least).
         assert!(capacity <= usize::MAX - max_leb128_len());
 
-        let file = File::create(path)?;
+        let file = File::options().read(true).write(true).create(true).truncate(true).open(path)?;
```

Comment:
Please add a comment explaining this. E.g. "Open the file for reading and writing, because some encoders do both (e.g. the metadata encoder when `-Zmeta-stats` is enabled)."